c7n/resources/account.py

```diff
@@ -248,7 +248,8 @@ class CloudTrailEnabled(Filter):
            'kms': {'type': 'boolean'},
            'kms-key': {'type': 'string'},
            'include-management-events': {'type': 'boolean'},
-           'log-metric-filter-pattern': {'type': 'string'}})
+           'log-metric-filter-pattern': {'type': 'string'},
+           'op': {'type': 'string', 'enum': list(OPERATORS.keys())}})
```

Comment:
Fair point about a straight string match being a bit limiting here :+1: . I think a top-level `op` key is likely to cause confusion though, especially given the number of top-level keys this filter already has.

That said, we do occasionally see a similar pattern elsewhere (in the security group [ingress filter](https://cloudcustodian.io/docs/aws/resources/security-group.html#aws-security-group-filters-ingress) for example) that serves the same purpose. We let a value be a simple string _or_ a nested value filter. So with a schema definition [something like this](https://github.com/cloud-custodian/cloud-custodian/blob/c8345c2c850da7a105c3beb28a80b3088920646d/c7n/resources/vpc.py#L1630-L1632) for `log-metric-filter-pattern` we might support both of these forms:

```yaml
    filters:
      - type: check-cloudtrail
        running: true
        log-metric-filter-pattern: "{ ($.eventName = \\"ConsoleLogin\\") }"
```

```yaml
    filters:
      - type: check-cloudtrail
        running: true
        log-metric-filter-pattern:
          type: value
          op: regex
          value: '\\{ ?(\\()? ?\\$\\.eventName ?= ?(")?ConsoleLogin(")? ?(\\))? ?&& ?'
```

Where it's a bit more explicit about what the `op` refers to.

Reply:
And this:
https://github.com/cloud-custodian/cloud-custodian/blob/d8a9452100681007d455b736d1b9b57d921d8a5b/c7n/resources/account.py#L267-L269
https://github.com/cloud-custodian/cloud-custodian/blob/d8a9452100681007d455b736d1b9b57d921d8a5b/c7n/resources/account.py#L335-L340